src/main/kotlin/io/github/mojira/arisa/modules/ThumbnailModule.kt

```diff
@@ -0,0 +1,189 @@
+     */
+    class LimitingInputStream(private val stream: InputStream, private val maxBytes: Long) : InputStream() {
+        private var totalReadAmount: Long = 0
+
+        private fun getRemaining(): Long = maxBytes - totalReadAmount
+
+        private fun getMaxReadAmount(desiredReadAmount: Int): Int {
+            if (desiredReadAmount <= 0) {
+                return 0
+            }
+
+            val remaining = getRemaining()
+            if (remaining <= 0) {
+                throw IOException("Trying to read more than $maxBytes bytes")
+            }
+            // Conversion to Int here is safe because result will be at most desiredReadAmount (= Int)
+            return min(remaining, desiredReadAmount.toLong()).toInt()
+        }
+
+        override fun read(): Int {
+            getMaxReadAmount(1) // Check that reading 1 byte is allowed
+            val result = stream.read()
+            if (result != -1) {
+                totalReadAmount++
+            }
+            return result
+        }
+
+        override fun read(b: ByteArray, off: Int, len: Int): Int {
+            val actualLen = getMaxReadAmount(len)
+            val readAmount = stream.read(b, off, actualLen)
+            if (readAmount != -1) {
+                totalReadAmount += readAmount
+            }
+            return readAmount
+        }
+
+        override fun available(): Int {
+            // Conversion to Int here is safe because result will be at most stream.available() (= Int)
+            return min(stream.available().toLong(), getRemaining()).toInt()
+        }
+
+        override fun close() {
+            stream.close()
+        }
+    }
+
+    private val log: Logger = LoggerFactory.getLogger(ThumbnailModule::class.java)
+
+    override fun invoke(issue: Issue, lastRun: Instant): Either<ModuleError, ModuleResponse> = with(issue) {
+        Either.fx {
+            var performedUpdate = false
+
+            // Only consider new issues
+            if (issue.created.isAfter(lastRun)) {
+                val oldDescription = issue.description
+                if (oldDescription != null) {
+                    val newDescription = replaceEmbeddedImages(issue, oldDescription)
+                    if (oldDescription != newDescription) {
+                        issue.updateDescription(newDescription)
+                        performedUpdate = true
+                    }
+                }
+            }
+
+            // Only consider new comments
+            val newComments = comments.filter {
+                it.created.isAfter(lastRun)
+            }
+
+            newComments.forEach {
+                val oldBody = it.body!!
+                val newBody = replaceEmbeddedImages(issue, oldBody)
+                if (oldBody != newBody) {
+                    it.update(newBody)
+                    performedUpdate = true
+                }
+            }
+
+            // TODO: Not sure if this is correct; should assert be called before
```

Comment:
```suggestion
            // TODO: Not sure if this is correct, should assert be called before
```
Detekt getting angered over the semicolon here.

Reply:
Thanks. I only added that as temporary comment to clear whether usage of assert is correct there. Have removed the comment now.